spynnaker/pyNN/models/neuron/structural_plasticity/synaptogenesis/formation/distance_dependent_formation.py

```diff
@@ -146,6 +152,11 @@ def distance(self, x0, x1, metric):
     @overrides(AbstractFormation.write_parameters)
     def write_parameters(self, spec):
         spec.write_array(self.__grid)
+        # Work out the reciprocal, but zero them if >= 1 as they are not
+        # representable as S031 in that case, and not used anyway
+        recip = 1 / self.__grid
+        recip[recip >= 1.0] = 0
+        spec.write_array(DataType.S031.encode_as_numpy_int_array(recip))
```

Comment:
id argue that recip is a param that you should do at init. so its just a write_array here.

Reply:
Not sure I agree; this allows grid to be changed if desired (yes I know it doesn't just now but I could see a future change where it becomes settable).  I don't think this is called more than once per run so not a big deal to do it here.

Reply:
i had the same debate after posting it. so fine to leave it be.